SmileAuth/Classes/SmileAuthenticator.m

```diff
@@ -304,4 +304,7 @@ +(void)clearPassword{
     [[SmileAuthenticator sharedInstance].keychainWrapper writeToKeychain];
 }
 
+-(void)changeAuthentication:(BOOL)newAuth {
+    _isAuthenticated = newAuth;
+}
 @end
```

Comment:
My use case involved using Touch ID to authenticate for one screen in my app, namely a payment info screen. However, after visiting the page one time without this method after navigating away the library would still think the user was authenticated. I added this method so that it could be called when navigating away from a certain screen if only that screen needed authentication. I went through the code and realized this was only called when the app was going to the background, and that wasn't enough for me. Adding this method expands the library to be used for single screens within an app, as well as for whole app authentication like the original version. 

> On Apr 15, 2016, at 9:41 PM, Rain <notifications@github.com> wrote:
> 
> In SmileAuth/Classes/SmileAuthenticator.m:
> 
> > @@ -304,4 +304,7 @@ +(void)clearPassword{
> >      [[SmileAuthenticator sharedInstance].keychainWrapper writeToKeychain];
> >  }
> >  
> > +-(void)changeAuthentication:(BOOL)newAuth {
> > +    _isAuthenticated = newAuth;
> > +}
> >  @end
> Could you tell me where & when this method is called? In ViewDidAppear or other case?
> 
> Because SmileAuthenticator observe UIApplicationDidEnterBackgroundNotification, add call below method:
> 
> -(void)appDidEnterBackground:(NSNotification*)notification{
>     if (_isAuthenticated) {
>         _didReturnFromBackground = YES;
>         _isAuthenticated = NO;
>     }
> }
> so even manually change _isAuthenticated, this property will be changed when app did enter background.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>